### PR TITLE
[Merged by Bors] - chore(UniformSpace.Basic): make `UniformSpace.comap` reducible 

### PR DESCRIPTION
**Pull request by mattrobball**
`UniformSpace.comap` is used in instance construction so needs to be reducible for unification purposes.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Topology/UniformSpace/Basic.lean [hunk 1]
```diff
@@ -1259,6 +1259,7 @@ instance inhabitedUniformSpaceCore : Inhabited (UniformSpace.Core α) :=
 
 /-- Given `f : α → β` and a uniformity `u` on `β`, the inverse image of `u` under `f`
   is the inverse image in the filter sense of the induced function `α × α → β × β`. -/
+@[reducible]
```
Comment (eric-wieser) [inline Mathlib/Topology/UniformSpace/Basic.lean:1262]: Can you add a reference to "reducible non-instances" here?

Comment (mattrobball) [reply Mathlib/Topology/UniformSpace/Basic.lean:1263]: Done

**Review by eric-wieser — APPROVED** [review]
bors merge

**Pull request closed without merging**